pkgs/development/tools/documentation/doxyrest/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchgit, cmake, lib, expat, ragelDev, which, lua5 }:
+
+stdenv.mkDerivation rec {
+  name = "doxyrest";
+  # Somehow fetchFromGithub was ignoring submodules
+  src = fetchgit {
+    url = "https://github.com/vovkos/doxyrest_b";
+    rev = "3500bc0101c5ebe4921d8cca49b42e209e761971";
+    sha256 = "1h1r0g9yklgf2f8qv4dw0lnsldpcr5rch14q934fvxqbpjh9b7fn";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ expat lua5 ragelDev which cmake ];
+
+  # otherwise "cc1: error: -Wformat-security ignored without -Wformat [-Werror=format-security]"
+  hardeningDisable = [ "format" ];
+
+  meta = with stdenv.lib; {
+    description = "A compiler from Doxygen XML to reStructuredText";
```

Comment:
Upstream says:

> This is a repository bundle (Doxyrest + AXL). With this bundle you can conveniently build Doxyrest and all the libraries it depends on -- in correct order and from the single source tree. 

Sounds like they try to solve a problem Nix already does.

Reply:
It's actually the build repo for https://github.com/vovkos/doxyrest
I guess in a sense it might be better to add a package for each sub-module and then finally open one for this package, but it seemed like an over-exuberant response at the moment, since I can find no (obvious) other use-cases for the submodules.

Reply:
I see. What happens when you try to build from https://github.com/vovkos/doxyrest directly?

Reply:
Nothing much, except it would require building each submodule at the commit referenced in the build directory. I could probably set that up over the weekend, shouldn't be hard, they seem to be `cmake` build stuff.

Reply:
The fact they "recommend" building with a totally separate repo might be suitable for users of other distros, but not for NixOS IMO. I think you should create derivations for each such submodule dependency, and try to tell the builder to use the "system"'s dependencies. See similar discussion at https://github.com/NixOS/nixpkgs/issues/91930#issuecomment-696601787 .

Reply:
That kind of defeats the purpose of having `fetchSubmodules` as an option for `fetchgit` then, would it not?

Reply:
I do not believe it makes sense to manually track the build repo (by checking which commits the submodules are pinned to) every time, just because "it feels more Nix like". There are reasons why we have support for `fetchSubmodules` and this is a good example IMO.

Reply:
I'm not against the usage of `fetchSubmodules`. I just think that the submodules should be exposed as Nix derivations, because yes, someone may find use to them in the future, and if they produce shared libraries they should be shared in the Nix Store.

> by checking which commits the submodules are pinned to) every time

You could write an `updateScript`, that will fetch the matching commits of these submodules, given a `rev` of the main repo, and then write a json / Nix file that will be used by the expression of https://github.com/vovkos/doxyrest .